File: frontend/src/api/types/chat-groups.ts
export interface ChatGroupRead {
  id: string;
  name: string;
  owner_user_id: string;
  character_id: string;
  selected_model_id: string;
  default_temperature: number;
  max_context_messages: number;
  auto_compaction_enabled: boolean;
  external_platform: string | null;
  external_group_id: string | null;
  external_account_id: string | null;
  created_at: string;
}

export interface ChatGroupPayload {
  name: string;
  character_id: number;
  selected_model_id: number;
  default_temperature?: number | null;
  max_context_messages?: number | null;
  auto_compaction_enabled?: boolean | null;
  external_platform?: string | null;
  external_group_id?: string | null;
  external_account_id?: string | null;
  initial_member_ids?: string[];
}

export interface ChatGroupUpdatePayload {
  name?: string;
  character_id?: number | null;
  selected_model_id?: number | null;
  default_temperature?: number | null;
  max_context_messages?: number | null;
  auto_compaction_enabled?: boolean | null;
  external_platform?: string | null;
  external_group_id?: string | null;
  external_account_id?: string | null;
}

export interface ChatGroupMemberRead {
  id: string;
  room_id: string;
  user_id: string;
  member_role: "admin" | "member" | string;
  created_at: string;
}

export interface ChatGroupStateRead {
  id: string;
  cocoon_id: string | null;
  chat_group_id: string;
  relation_score: number;
  persona_json: Record<string, unknown>;
  active_tags_json: string[];
  current_wakeup_task_id: string | null;
}

File: backend/app/schemas/workspace/chat_groups.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel
from app.schemas.workspace.cocoons import ChatMessageCreate


class ChatGroupRoomCreate(BaseModel):
    name: str = Field(min_length=1)
    character_id: str
    selected_model_id: str
    default_temperature: float | None = None
    max_context_messages: int | None = Field(default=None, ge=1)
    auto_compaction_enabled: bool | None = None
    external_platform: str | None = None
    external_group_id: str | None = None
    external_account_id: str | None = None
    initial_member_ids: list[str] = Field(default_factory=list)


class ChatGroupRoomUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1)
    character_id: str | None = None
    selected_model_id: str | None = None
    default_temperature: float | None = None
    max_context_messages: int | None = Field(default=None, ge=1)
    auto_compaction_enabled: bool | None = None
    external_platform: str | None = None
    external_group_id: str | None = None
    external_account_id: str | None = None


class ChatGroupRoomOut(ORMModel):
    id: str
    name: str
    owner_user_id: str
    character_id: str
    selected_model_id: str
    default_temperature: float
    max_context_messages: int
    auto_compaction_enabled: bool
    external_platform: str | None
    external_group_id: str | None
    external_account_id: str | None
    created_at: datetime


class ChatGroupMemberCreate(BaseModel):
    user_id: str
    member_role: str = "member"


class ChatGroupMemberUpdate(BaseModel):
    member_role: str


class ChatGroupMemberOut(ORMModel):
    id: str
    room_id: str
    user_id: str
    member_role: str
    created_at: datetime


class ChatGroupStateOut(ORMModel):
    id: str
    cocoon_id: str | None = None
    chat_group_id: str
    relation_score: int
    persona_json: dict
    active_tags_json: list[str]
    current_wakeup_task_id: str | None


class ChatGroupMessageCreate(ChatMessageCreate):
    pass


class MessageRetractResult(ORMModel):
    message_id: str
    is_retracted: bool
    retracted_at: datetime | None
    retracted_by_user_id: str | None
    retraction_note: str | None
